plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout == 0 {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	s.endpointMap = map[string]bool{}
+	for _, endpoint := range s.IgnoredEndpoints {
+		s.endpointMap[strings.ToLower(endpoint)] = true
+	}
+
+	if s.URL == "" {
+		return errors.New("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("empty hostname for url %q", s.URL)
+	}
+
+	s.baseURL = u
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("invalid scheme %q", u.Scheme)
+	}
+
+	var tlsCfg *stdTls.Config
+	if u.Scheme == "https" {
+		tlsCfg, err = s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = u.Host
+	configuration.Scheme = u.Scheme
+	configuration.UserAgent = internal.ProductToken()
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	s.client = goslurm.NewAPIClient(configuration)
+
+	return nil
+}
+
+func (s *Slurm) parseTres(tres string) map[string]interface{} {
+	parsedValues := map[string]interface{}{}
+	var ok bool
+
+	for _, tresVal := range strings.Split(tres, ",") {
+		parsedTresVal := strings.Split(tresVal, "=")
+		if len(parsedTresVal) != 2 {
+			continue
+		}
+
+		tag := parsedTresVal[0]
+		val := parsedTresVal[1]
+		var factor float64 = 1
+
+		if tag == "mem" {
+			factor, ok = map[string]float64{
+				"K": 1.0 / 1024.0,
+				"M": 1,
+				"G": 1024,
+				"T": 1024 * 1024,
+			}[strings.ToUpper(val[len(val)-1:])]
+			if !ok {
+				continue
+			}
+			val = val[:len(val)-1]
+		}
+
+		parsedInt, err := strconv.ParseInt(val, 10, 64)
+		if err == nil && factor >= 1 {
+			parsedValues[tag] = parsedInt * int64(factor)
+			continue
+		}
+		parsedFloat, err := strconv.ParseFloat(val, 64)
+		if err == nil {
+			parsedValues[tag] = parsedFloat * factor
+			continue
+		}
+		parsedValues[tag] = val
+	}
+
+	return parsedValues
+}
+
+func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator,
+	diag *goslurm.V0038DiagStatistics) {
+	records := make(map[string]interface{})
+	tags := make(map[string]string)
+
+	var (
+		int32Ptr *int32
+		boolPtr  *bool
+		ok       bool
+	)
+
+	tags["source"] = s.baseURL.Hostname()
+
+	if int32Ptr, ok = diag.GetServerThreadCountOk(); ok {
+		records["server_thread_count"] = *int32Ptr
+	}
```

Comment:
```suggestion
	if int32Ptr, ok := diag.GetServerThreadCountOk(); ok {
		records["server_thread_count"] = *int32Ptr
	}
```

Same for the other `if` statements below.

Reply:
Done on 47393078166c438b5a0f2263661b1170b5c02f67.